src/DacpacTool/ScriptParser.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Linq;
+using System.Reflection;
+using Microsoft.SqlTools.ServiceLayer.BatchParser;
+
+namespace MSBuild.Sdk.SqlProj.DacpacTool
+{
+    public class ScriptParser : ICommandHandler
+    {
+        private readonly Parser _parser;
+        private bool _parsed;
+        private readonly List<string> _includedFileNames = new List<string>();
+        private StringBuilder _scriptBuilder = new StringBuilder();
+        private string _currentFilePath;
+
+        public ScriptParser(string sourceFile, IVariableResolver variableResolver)
+        {
+            if (Path.IsPathRooted(sourceFile))
+            {
+                _currentFilePath = Path.GetDirectoryName(sourceFile);
+            }
+            else
+            {
+                sourceFile = Path.Combine(Environment.CurrentDirectory, sourceFile);
+                _currentFilePath = Path.GetDirectoryName(sourceFile);
+            }
+
+            _parser = new Parser(this, variableResolver, new StreamReader(sourceFile), sourceFile);
+        }
+
+        protected void Parse()
```

Comment:
Shouldn't this be private?

Reply:
I think I wrote this in a moment of irritation with being unable to access any of the members of `Microsoft.SqlTools.ServiceLayer.BatchParser.Parser`. 😄  I'll set it to private.

Reply:
Set to private.